inf-ruby.el

```diff
@@ -505,10 +505,131 @@ Must not contain ruby meta characters.")
+          (skip-chars-backward "\r\n[:blank:]")
+          (let* ((beg (if (consp where)
+                          (car where)
+			(save-excursion
+                          (backward-sexp 1)
+                          (point))))
+		 (end (if (consp where)
+                          (cdr where)
+			(line-end-position)))
+		 (display-string (format format value))
+		 (o nil))
+            (remove-overlays beg end 'category type)
+            (funcall #'put-text-property
+                     0 (length display-string)
+                     'face prepend-face
+                     display-string)
+            ;; If the display spans multiple lines or is very long, display it at
+            ;; the beginning of the next line.
+            (when (or (string-match "\n." display-string)
+                      (> (string-width display-string)
+			 (- (window-width) (current-column))))
+              (setq display-string (concat " \n" display-string)))
+            ;; Put the cursor property only once we're done manipulating the
+            ;; string, since we want it to be at the first char.
+            (put-text-property 0 1 'cursor 0 display-string)
+            (when (> (string-width display-string) (* 3 (window-width)))
+              (setq display-string
+                    (concat (substring display-string 0 (* 3 (window-width)))
+                            "...\nResult truncated.")))
+            ;; Create the result overlay.
+            (setq o (apply #'inf-ruby--make-overlay
+                           beg end type
+                           'after-string display-string
+                           props))
+            (pcase duration
+              ((pred numberp) (run-at-time duration nil #'inf-ruby--delete-overlay o))
+              (`command (if this-command
+                            (add-hook 'pre-command-hook
+                                      #'inf-ruby--remove-result-overlay
+                                      nil 'local)
+                          (inf-ruby--remove-result-overlay))))
+            (let ((win (get-buffer-window buffer)))
+              ;; Left edge is visible.
+              (when (and win
+			 (<= (window-start win) (point))
+			 ;; In 24.3 `<=' is still a binary predicate.
+			 (<= (point) (window-end win))
+			 ;; Right edge is visible. This is a little conservative
+			 ;; if the overlay contains line breaks.
+			 (or (< (+ (current-column) (string-width value))
+				(window-width win))
+                             (not truncate-lines)))
+		o))))))))
+
+(defun inf-ruby--remove-result-overlay ()
+  "Remove result overlay from current buffer.
+This function also removes itself from `pre-command-hook'."
+  (remove-hook 'pre-command-hook #'inf-ruby--remove-result-overlay 'local)
+  (remove-overlays nil nil 'category 'result))
+
+(defun inf-ruby--eval-overlay (value)
+  "Make overlay for VALUE at POINT."
+  (inf-ruby--make-result-overlay (format "%S" value) (point) 'command)
+  value)
+
+
```

Comment:
An unnecessary newline.

Reply:
Ops I've removed on this commit https://github.com/nonsequitur/inf-ruby/pull/153/commits/c49b3d360a28ea670e92d56d64010cd97bdbc351 🙈